sootup.java.core/src/main/java/sootup/java/core/interceptors/typeresolving/TypePromotionVisitor.java

```diff
@@ -56,6 +56,13 @@ public static boolean isIntermediateType(Type type) {
   }
 
   public void visit(@Nonnull Value value, @Nonnull Type stdType, @Nonnull Stmt stmt) {
```

Comment:
can we be more specific?

```suggestion
  public void visit(@Nonnull Local value, @Nonnull Type stdType, @Nonnull Stmt stmt) {
```

Reply:
we are calling visit function with many different types JInstanceOfExpr, JNewArrayExpr etc so changing in parameter directly won't solve.

if (value instanceof Constant) {return;} I tried this as well instead of [if (!(value instanceof Local))] with different jars but seems like no other special case than Constant so could keep just the check for Constant as well

Reply:
regarding 1st point, I found a TC where if we don't promote just constants, it would lead to an invalid type for int1 type to boolean. so reverted